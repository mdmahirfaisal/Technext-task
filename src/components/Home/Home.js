import React from 'react';
import { CircularProgress } from '@mui/material';
import DisplayAllData from '../DisplayAllData/DisplayAllData';

const Home = ({ displayData, loading }) => {

    return (
        <div data-testid="myhomediv">
            {loading ?
                <h1 className='mt-12 font-bold'><CircularProgress /> </h1>
                :
                <div className="grid grid-cols-1 sm:grid-cols-2 lg:grid-cols-3 xl:grid-cols-4 gap-2 sm:gap-3 my-12">

                    {displayData?.map((launch, index) => <DisplayAllData
                        key={index}
                        launch={launch}
                    />)}

                </div>
            }
        </div>
    );
};

export default Home;