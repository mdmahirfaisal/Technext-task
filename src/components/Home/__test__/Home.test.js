import React from 'react'
import ReactDOM from 'react-dom'
import Home from '../Home'
import renderer from 'react-test-renderer'


it('Home render without crashing',
    () => {
        const div = document.createElement('div')
        ReactDOM.render(<Home />,
            div)
    })

it("matchs Home snapshot",
    () => {
        const tree = renderer.create(<Home />).toJSON()
        expect(tree).toMatchSnapshot()
    })