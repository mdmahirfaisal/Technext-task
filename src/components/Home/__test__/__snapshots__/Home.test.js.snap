// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matchs Home snapshot 1`] = `
<div
  data-testid="myhomediv"
>
  <button>
    test button
  </button>
  <div
    className="grid grid-cols-1 sm:grid-cols-2 lg:grid-cols-3 xl:grid-cols-4 gap-2 sm:gap-3 my-12"
  />
</div>
`;
